packages/assets-controllers/src/NftDetectionController.ts

```diff
@@ -300,6 +323,116 @@ export class NftDetectionController extends BaseController<
    */
   isMainnet = (): boolean => this.config.chainId === ChainId.mainnet;
 
+  isMainnetByNetworkClientId = (networkClient: NetworkClient): boolean => {
+    return networkClient.configuration.chainId === ChainId.mainnet;
+  };
+
+  #selectedAddressByNetworkClientId = new Map<NetworkClientId, string>();
+
+  override startPollingByNetworkClientId(
+    networkClientId: string,
+    selectedAddress?: string,
+  ): string {
+    // selectedAddress needs to be optional in the interface for the override to work with the mixin
```

Comment:
is this the only case we see needing to extend the PollingController with extra data for the networkClientId?

Reply:
Probably not. We will want to explore how an enhancement of the pollingController could serve these cases more elegantly: https://github.com/MetaMask/MetaMask-planning/issues/1406